tracer/src/Datadog.Trace/Debugger/Caching/DefaultMemoryChecker.cs

```diff
@@ -0,0 +1,166 @@
+// <copyright file="DefaultMemoryChecker.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.Caching;
+
+internal class DefaultMemoryChecker : IMemoryChecker
+{
+    private const long LowMemoryThreshold = 1_073_741_824; // 1 GB in bytes
+
+    private static readonly IDatadogLogger Logger = DatadogLogging.GetLoggerFor<DefaultMemoryChecker>();
+
+    private static readonly Lazy<DefaultMemoryChecker> _instance = new Lazy<DefaultMemoryChecker>(() => new DefaultMemoryChecker(), LazyThreadSafetyMode.ExecutionAndPublication);
+
+    private readonly bool _isLowResourceEnvironment;
+
+    private DefaultMemoryChecker()
+    {
+        _isLowResourceEnvironment = CheckLowResourceEnvironment();
+    }
+
+    public static DefaultMemoryChecker Instance => _instance.Value;
+
+    [return: MarshalAs(UnmanagedType.Bool)]
+    [DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+    private static extern bool GlobalMemoryStatusEx([In, Out] MEMORYSTATUSEX lpBuffer);
+
+    public bool IsLowResourceEnvironment()
```

Comment:
Should this be a property instead of a method?

Reply:
Thanks! Fixed in 93ce326e53379d4935cf896c25e5600ea89b4164